### ISSUE TITLE
Invariant Violation: Invalid hook call. Hooks can only be called inside of the body of a function component

### ISSUE THREAD

**Issue by codeshaker** [facebook/react#15628]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

Root.js
```
import React from "react";
import { Provider } from "react-redux";
import { Navigator } from "./Navigation/Index";
import { store } from "./Store/Index";

export default class Root extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <Navigator />
      </Provider>
    );
  }
}
```

Store.js
```
import ReduxThunk from "redux-thunk";
import { createStore, applyMiddleware, compose } from "redux";
import reducer from "../Reducers/index";

let composeEnhancers = compose;
/*  eslint no-undef: 0    */
if (__DEV__) {
  /*  eslint no-underscore-dangle: 0    */
  composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
}

const store = createStore(
  reducer,
  {},
  composeEnhancers(applyMiddleware(ReduxThunk))
);

export { store };
```

LoginScreen.js
```
import React, { Component } from "react";
import PhoneInput from "react-native-phone-input";
import { connect } from "react-redux";
import { View, StatusBar } from "react-native";
import { Container, Item, Input, Button, Text } from "native-base";
import {
  phoneChanged,
  codeChanged,
  onCodeDispatched,
  onPhoneLogin,
  clearAuth,
  onSignOut
} from "../Actions/AuthActions";
//import firebase from "react-native-firebase";
import { auth } from "../Config/firebase";

export class LoginScreen extends Component {
}

export default connect(
  null, // passing null just for testing
  null
)(LoginScreen);
```

package.json
```
 "dependencies": {
    "@babel/runtime": "^7.4.4",
    "firebase": "^5.11.1",
    "isomorphic-fetch": "^2.2.1",
    "native-base": "^2.12.1",
    "prop-types": "^15.7.2",
    "react": "^16.8.6",
    "react-native": "0.57.5",
    "react-native-firebase": "^5.3.1",
    "react-native-gesture-handler": "^1.1.0",
    "react-native-otp-inputs": "^2.0.1",
    "react-native-phone-input": "^0.2.1",
    "react-navigation": "^3.9.1",
    "react-redux": "^7.0.3",
    "redux": "^4.0.1",
    "redux-immutable-state-invariant": "^2.1.0",
    "redux-thunk": "^2.3.0"
  },
  "devDependencies": {
    "babel-polyfill": "^6.26.0",
    "babel-preset-expo": "^5.0.0",
    "eslint-plugin-react-hooks": "^1.6.0",
    "expo": "^32.0.6"
  },
```
**Code Base**

https://codesandbox.io/s/github/codeshaker/HomeClub/tree/master/FrontEnd/HomeClubUI

**What is the expected behavior?**

**I am getting Error when trying to connect my LoginScreen Component with Redux Store using connect() Function of Store.** For testing purpose i am passing null for both parameters but getting the same errors always.

I am not able to debug when its causing this error. I am not breaking any of:

    You might have mismatching versions of React and the renderer (such as React DOM)
    You might be breaking the Rules of Hooks
    You might have more than one copy of React in the same app

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**

**Comment by threepointone** [comment]
This is hard to debug with a reproducible version, but could you try replacing -
```jsx
export class LoginScreen extends Component {
}
```
with 
```jsx
export class LoginScreen extends Component {
  render() { return null }
}
```
and see if the behaviour changes?

if the error persists, I recommend you post at https://github.com/reduxjs/react-redux/issues (or reproduce the bug without dependencies to isolate that this is actually a react issue)

**Comment by codeshaker** [comment]
I Tried your changes but did not work. I have create a sandbox project which will help you reproduce the code.

**Comment by codeshaker** [comment]
Any help will be deeply appreciated.

**Comment by ir-fuel** [comment]
@threepointone I have managed to reproduce it in a tiny project with a minimum of dependencies

https://github.com/ir-fuel/material-test

Just run `npm run development` and have a webserver pointing at `dist` and open `index.html`

**Comment by ir-fuel** [comment]
Here is the index.html + bundle.js (built with `eval`devtool)

[Archive.zip](https://github.com/facebook/react/files/3218484/Archive.zip)

**Comment by codeshaker** [comment]
Actually its a version issue reducing react-redux package version from 7 to 6 (which is compatible with react native version 0.57) solved the issue. Got to knew it yesterday.

Closing it.

**Comment by ir-fuel** [comment]
My example does not use redux

**Comment by dotansimha** [comment]
We had a very similar issue in https://github.com/dotansimha/graphql-code-generator . 
I found out that this caused it in our case:
- We are using Yarn Workspaces for monorepo, and we had the following structure:
```
packages/
  - a <- Part of the workspace, NodeJS library
  - b <- Part of the workspace, NodeJS library
  - c <- Part of the workspace, NodeJS library
  - website <- NOT Part of the workspace, and has a dependency for React
     - live-demo <- Part of the workspace, React app
```

The `website` dir is not part of the workspace, but when trying to run `live-demo`, it causes the same error you're having. If I add `website` to the Workspace, it seems to work. If I remove manually `website/node_modules/react` and  `website/node_modules/react-dom` - everything works.

So basically, something causes React to think that there are 2 different copies of React/ReactDOM (as mentioned in the error).

**Comment by edwinjue** [comment]
I was receiving the error after I added `HtmlWebpackPlugin` to my project. Little did I know, `HtmlWebpackPlugin` will inject `bundle.js` to the header of dist/index.html. 

So, if you're getting this error and use `HtmlWebpackPlugin` check the index.html in your output directory and remove  `<script src="./bundle.js"></script>` if you added it explicitly.  Make sure `bundle.js` not showing up twice when you `view-source:` the site like so:

![image](https://user-images.githubusercontent.com/1448719/198854535-fa87e996-1307-414e-8261-ed7903192f79.png)

Hope this helps somebody else!
